flux/src/test/java/com/groupbyinc/util/UrlFunctionsBugTest.java

```diff
@@ -0,0 +1,118 @@
+package com.groupbyinc.util;
+
+import com.groupbyinc.api.model.Navigation;
+import com.groupbyinc.api.model.refinement.RefinementValue;
+import com.groupbyinc.api.tags.UrlFunctions;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.servlet.jsp.JspException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by ferron on 4/22/15.
+ */
+public class UrlFunctionsBugTest {
+
+    private final String DEFAULT_BEAUTIFIER = "default";
+    private final String REFINEMENT_REGEX = "~((?=[\\w]*[=:]))";
+
+    UrlBeautifier urlBeautifier;
+
+    @Before
+    public void setUp() {
+        UrlBeautifier.INJECTOR.set(new HashMap<String, UrlBeautifier>());
+        UrlBeautifier.createUrlBeautifier(DEFAULT_BEAUTIFIER);
+        urlBeautifier = UrlBeautifier.getUrlBeautifiers().get(DEFAULT_BEAUTIFIER);
+        urlBeautifier.addRefinementMapping('s', "size");
+        urlBeautifier.setSearchMapping('q');
+        urlBeautifier.setAppend("/index.html");
+        urlBeautifier.addReplacementRule('/', ' ');
+        urlBeautifier.addReplacementRule('\\', ' ');
+    }
+
+    @Test
+    public void refinementAdditionWithMapping() throws JspException, AbstractUrlBeautifier.UrlBeautificationException {
+
+        List<Navigation> navigations = new ArrayList<Navigation>();
+
+        String refinementString = "Category Root~Athletics~Men's~Sneakers";
+
+        String url = UrlFunctions.toUrlAdd(
+                DEFAULT_BEAUTIFIER, "", navigations, "category_leaf_expanded", new RefinementValue().setValue(
+                        refinementString).setCount(5483));
+        assertEquals(
+                "/index.html?refinements=%7Ecategory_leaf_expanded%3DCategory+Root%7EAthletics%7EMen%27s%7ESneakers",
+                url);
+    }
+
+    @Test
```

Comment:
This test actually tests REGEX, not the class UrlBeautifier.  If someone changes the class UrlBeautifier this test will never break and is useless from protecting us against other developers.

I think you should add a method to the UrlBeautifier that does the split logic, make it package private and have this test call that method.